### PR TITLE
Remove `store` & `store/orm`

### PR DESCRIPTION
**Pull request by archseer**
Depends on https://github.com/smartcontractkit/chainlink/pull/4994

### CODE REVIEW

File: core/cmd/client.go [hunk 1]
```diff
@@ -75,24 +80,66 @@ type ChainlinkAppFactory struct{}
...
 // NewApplication returns a new instance of the node with the given config.
 func (n ChainlinkAppFactory) NewApplication(cfg config.GeneralConfig) (chainlink.Application, error) {
-	advisoryLocker := postgres.NewAdvisoryLock(cfg.DatabaseURL())
 	shutdownSignal := gracefulpanic.NewSignal()
-	// TODO: Remove store entirely
-	// https://app.clubhouse.io/chainlinklabs/story/12980/remove-store-object-entirely
-	store, err := strpkg.NewStore(cfg, advisoryLocker, shutdownSignal)
+	uri := cfg.DatabaseURL()
+	dialect := cfg.GetDatabaseDialectConfiguredOrDefault()
+	db, gormDB, err := postgres.NewConnection(uri.String(), string(dialect), cfg, shutdownSignal)
```
Comment (samsondav) [inline core/cmd/client.go:86]: This is pulled up and out in a later PR I would assume?

File: core/cmd/client.go [hunk 2]
```diff
@@ -75,24 +80,66 @@ type ChainlinkAppFactory struct{}
...
-	sqlxDB := postgres.UnwrapGormDB(db)
-	cfg.SetDB(db)
-	keyStore := keystore.New(db, utils.GetScryptParams(cfg))
+	cfg.SetDB(gormDB)
+
+	// Set up the versioning ORM
+	verORM := versioning.NewORM(db)
+
+	// Set up periodic backup
+	if cfg.DatabaseBackupMode() != config.DatabaseBackupModeNone {
```
Comment (samsondav) [inline core/cmd/client.go:96]: Can we put this in a function or something? Like "setup DB"

File: core/cmd/client.go [hunk 3]
```diff
@@ -75,24 +80,66 @@ type ChainlinkAppFactory struct{}
...
+		if version != nil {
+			versionString = version.Version
+		}
+
+		databaseBackup := periodicbackup.NewDatabaseBackup(cfg, logger.Default)
+		databaseBackup.RunBackupGracefully(versionString)
+	}
+
+	// Migrate the database
+	if cfg.MigrateDatabase() {
```
Comment (samsondav) [inline core/cmd/client.go:120]: Feels really hokey to do all these mutative/destructive actions in `NewApplication`, but I guess it always did this and it was simply hidden

File: core/services/postgres/connection.go [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+	"github.com/smartcontractkit/sqlx"
+	gormpostgres "gorm.io/driver/postgres"
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+
+	// We've specified a later version in go.mod than is currently used by gorm
+	// to get this fix in https://github.com/jackc/pgx/pull/975.
+	// As soon as pgx releases a 4.12 and gorm [https://github.com/go-gorm/postgres/blob/master/go.mod#L6]
+	// bumps their version to 4.12, we can remove this.
```
Comment (samsondav) [inline core/services/postgres/connection.go:22]: is there a story to track?

File: core/services/postgres/log_wrapper.go [hunk 5]
```diff
@@ -1,4 +1,4 @@
-package orm
+package postgres
```
Comment (samsondav) [inline core/services/postgres/log_wrapper.go:1]: ❤️

File: core/services/postgres/connection.go (hunk 4 above)
Comment (archseer) [reply core/services/postgres/connection.go:22]: I think this is obsolete, I copied it verbatim from https://github.com/smartcontractkit/chainlink/blob/3fc7397b49d4b5b9a03c3fdf46e3bc256064e508/core/store/orm/orm.go#L24-L28

**Review by samsondav — DISMISSED** [review]

File: core/services/postgres/connection.go (hunk 4 above)
Comment (jmank88) [reply core/services/postgres/connection.go:22]: pgx updated to 4.13 in #5061

Comment (jmank88) [reply core/services/postgres/connection.go:22]: And go-gorm/postgres updated to 4.13 @ v1.1.1 ten days ago: https://github.com/go-gorm/postgres/blob/v1.1.1/go.mod

**Review by samsondav — APPROVED** [review]

File: core/store/migrate/migrations/0068_remove_unused_columns.sql [hunk 6]
```diff
@@ -0,0 +1,3 @@
+-- +goose Up
+ALTER TABLE external_initiators DROP COLUMN IF EXISTS deleted_at;
```
Comment (samsondav) [inline core/store/migrate/migrations/0068_remove_unused_columns.sql:2]: Do we want to hard-delete any records that have this set?

**Review by jmank88 — APPROVED** [review]

**Review by samsondav — APPROVED** [review]

**Pull request merged**